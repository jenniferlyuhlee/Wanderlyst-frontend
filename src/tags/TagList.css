.TagList{
    text-align: center;
}

.TagList-text{
    padding: 2rem;
    background-color: rgba(102, 209, 149);
    color: white;
}

.TagList-grid{
    display: grid;
    grid-template-columns: repeat(5, 1fr);
}

@media (max-width: 1000px){
    .TagList-grid{
        grid-template-columns: repeat(2, 1fr);
    }
}