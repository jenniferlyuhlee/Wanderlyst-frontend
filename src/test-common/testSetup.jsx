import React, { useState } from "react";
import UserContext from "../auth-user/UserContext";
import WanderlystApi from "../utils/api";
import '@testing-library/jest-dom';

const demoUser = {
  username: "testadmin", 
  firstName: "testfirst",
  lastName: "testlast",
  location: "testloc",
  bio: "testbio",
  profilePic: null,
  isAdmin: true,
};

const demoTags = [
  {id: 1, name: 'Foodie', description: 'Savor the world one bite at a time. Dive into a journey packed with delicious local cuisines, hidden culinary gems, and must-try dishes that will leave your taste buds tingling.'},
  {id: 2, name: 'Explorer', description: 'Step off the beaten path and uncover the unknown. From ancient ruins to vibrant cityscapes, let your curiosity guide you through unforgettable experiences.'}
]

const UserProvider = ({ children, 
                        currUser = demoUser, 
                        setCurrUser, 
                        tags = demoTags,
                        logout,
                        hasLikedItin,
                        toggleLike }) => {
  const [user, setUser] = useState(currUser);

  return (
    <UserContext.Provider value={{ currUser: user, 
                                  setCurrUser: setUser, 
                                  tags, logout, 
                                  hasLikedItin, toggleLike}}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider };