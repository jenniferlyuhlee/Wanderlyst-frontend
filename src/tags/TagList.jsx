import React, { useState, useEffect, useContext} from "react";
import WanderlystApi from "../utils/api";
import UserContext from "../auth-user/UserContext";
import TagCard from "./TagCard";
import Loading from "../shared/Loading";
import "./TagList.css"

function TagList(){
    const { tags } = useContext(UserContext);

    if(!tags){
        return <Loading />
    }

    return(
        <div className = "TagList">
            <div className="TagList-text">
                <p className = "fs-4 mb-4">Explore itineraries by tags</p>
                <p>On Wanderlyst, we categorize itineraries by 10 different tagsâ€”each corresponding 
                to a travel style and identity. Click on each tag to learn more.</p>
            </div>
            <div className = "TagList-grid">
                {tags.map(t => <TagCard key={t.id} tag = {t}/>)}
            </div>
        </div>

    )
}

export default TagList;