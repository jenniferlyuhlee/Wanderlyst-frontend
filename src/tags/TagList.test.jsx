import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { UserProvider } from "../test-common/testSetup";
import { MemoryRouter} from "react-router-dom";
import TagList from "./TagList";
import {expect, describe, test } from "vitest";

describe("TagList component", () => {
    test("renders without crashing", () => {
        render(
            <MemoryRouter >
                <UserProvider>
                    <TagList/>
                </UserProvider>
            </MemoryRouter>)
    });
    test("matches snapshot", () => {
        const{asFragment} = render(
            <MemoryRouter >
                <UserProvider>
                    <TagList/>
                </UserProvider>
            </MemoryRouter>)
        expect(asFragment()).toMatchSnapshot();
    });
    
    test("displays as expected: all tags in test setup", async () => {
        const{getByText} = render(
            <MemoryRouter >
                <UserProvider>
                    <TagList/>
                </UserProvider>
            </MemoryRouter>)
        
        await waitFor(() => {
            expect(getByText("Explore itineraries by tags")).toBeInTheDocument();
            expect(getByText("Foodie")).toBeInTheDocument();
            expect(getByText("Explorer")).toBeInTheDocument();
        });
    });

    test("matches snapshot", () => {
        const{asFragment} = render(
            <MemoryRouter >
                <UserProvider>
                    <TagList/>
                </UserProvider>
            </MemoryRouter>)
        expect(asFragment()).toMatchSnapshot();
    });
});