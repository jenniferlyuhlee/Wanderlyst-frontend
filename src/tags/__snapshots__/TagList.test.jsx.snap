// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagList component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TagList"
  >
    <div
      class="TagList-text"
    >
      <p
        class="fs-4 mb-4 wanderlyst"
      >
        Explore itineraries by tags
      </p>
      <p
        class="fw-semibold"
      >
        On Wanderlyst, we categorize itineraries by 10 different tags—each corresponding to a travel style and identity. Click on each tag to learn more.
      </p>
    </div>
    <div
      class="TagList-grid"
    >
      <a
        class="TagCard"
        href="/tags/Foodie"
      >
        <div
          class="Tag-bg tag-bg-Foodie"
        >
          <div
            class="TagCard-overlay"
          >
            <h4>
              Foodie
            </h4>
          </div>
        </div>
      </a>
      <a
        class="TagCard"
        href="/tags/Explorer"
      >
        <div
          class="Tag-bg tag-bg-Explorer"
        >
          <div
            class="TagCard-overlay"
          >
            <h4>
              Explorer
            </h4>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TagList component > matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="TagList"
  >
    <div
      class="TagList-text"
    >
      <p
        class="fs-4 mb-4 wanderlyst"
      >
        Explore itineraries by tags
      </p>
      <p
        class="fw-semibold"
      >
        On Wanderlyst, we categorize itineraries by 10 different tags—each corresponding to a travel style and identity. Click on each tag to learn more.
      </p>
    </div>
    <div
      class="TagList-grid"
    >
      <a
        class="TagCard"
        href="/tags/Foodie"
      >
        <div
          class="Tag-bg tag-bg-Foodie"
        >
          <div
            class="TagCard-overlay"
          >
            <h4>
              Foodie
            </h4>
          </div>
        </div>
      </a>
      <a
        class="TagCard"
        href="/tags/Explorer"
      >
        <div
          class="Tag-bg tag-bg-Explorer"
        >
          <div
            class="TagCard-overlay"
          >
            <h4>
              Explorer
            </h4>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
